### ISSUE TITLE
Can't get Kundera to skip over some @Entity objects

### ISSUE THREAD

**Issue by bytor99999** [Impetus/kundera#109]
I have many databases and I am using Kundera only for Cassandra. However, I also have Postgres in which I have @Entity mappings for some domain objects that are just for Postgres, which I am using Hibernate/Spring 3.1 and Spring Data JPA. For Cassandra I am using Kundera. So I had had two persistence units in my persistence.xml and set everything up. I am getting errors because of the entities that are for Postgres not having @Table on them. But those entities shouldn't be pickedup/scanned by Kundera, but it is and throwing exceptions. If I put the two Kundera entities in the persistence unit xml and exclude classes not in the xml, then the two classes I list in the xml aren't picked up and no MetaData is created for them, even though I correctly put them in the persistence.xml

here is the persistence unit definition

`<persistence-unit name="cassandra_pu">
        <provider>com.impetus.kundera.KunderaPersistence</provider>
        <class>com.myappname.domain.cassandra.Persons</class>
        <class>com.myappname.domain.cassandra.CassandraAddress</class>
        <exclude-unlisted-classes>true</exclude-unlisted-classes>
        <properties>
            <property name="kundera.nodes" value="10.0.30.74"/>
            <property name="kundera.port" value="9160"/>
            <property name="kundera.keyspace" value="testing"/>
            <property name="kundera.dialect" value="cassandra"/>
            <property name="kundera.client.lookup.class" value="com.impetus.client.cassandra.pelops.PelopsClientFactory" />
            
        </properties>
    </persistence-unit>`

The classes are 100% correct names and packages.

With that I get

```Caused by: java.lang.NullPointerException
    at com.impetus.kundera.persistence.event.EntityEventDispatcher.fireEventListeners(EntityEventDispatcher.java:53)
    at com.impetus.kundera.persistence.PersistenceDelegator.persist(PersistenceDelegator.java:134)
    at com.impetus.kundera.persistence.EntityManagerImpl.persist(EntityManagerImpl.java:279)
    ... 40 more

```

Thanks

Mark
```

**Comment by mevivs** [comment]
I guess it should work. Looking into this

**Comment by mevivs** [comment]
Using 2.0.7 release or older?

**Comment by mevivs** [comment]
Okay. I had a look. Something on same is reported earlier and have been fixed in kundera-2.0.7.1 branch. Here are the commit
7a9e4e7152958a05be8efb66e388e3d53d882664
and
9e46ae5fce9746fa82fd5aececb8dec6c534ed07

Please verify if it is working fine with 2.0.7.1 for you?

**Comment by bytor99999** [comment]
Using 2.0.7

**Comment by bytor99999** [comment]
Thanks

Mark

**Comment by bytor99999** [comment]
When will this get into a jar on a maven repo?

Thanks

Mark

**Comment by bytor99999** [comment]
Another stupid question. But when I see a commit like "Merge branch 'trunk' of github.com:impetus-opensource/Kundera into trunk" Is this merging the changes of the branch into trunk, or just rebasing the branch with changes in trunk? Basically, if I am forced to have to build from source because we need this feature now, should I build from trunk or from 2.0.7.1 branch?

Thanks

Mark

**Comment by mevivs** [comment]
Is it working for you? Not yet planned to release 2.0.7.1, as fixes are going on merged in trunk as well. 

Trunk is current development branch (2.0.8 release)and might be bit unstable. I would suggest to build 2.0.7.1.

**Comment by bytor99999** [comment]
Thanks, I did build from 2.0.7.1 branch. However, when I use mvn install,
it seems to get a bunch of stuff from 2.0.8.SNAPSHOT.

I am only building core and cassandra modules, since those are all that I
need.

And when I then run the test from my project I get a Class not found
exception on

Caused by: java.lang.ClassNotFoundException:
com.impetus.kundera.KunderaPersistence

And I know that class exists in the jar that got into my .m2 repo for
2.7.0.1, so my IDE IDEA must not put it into the classpath using the
pom.xml for my project even though I have



<dependency
    <groupId>com.impetus.client</groupId>
    <artifactId>kundera-cassandra</artifactId>
    <version>${kundera.version}</version>
</dependency>

and property kundera.version is set to 2.0.7.1

Ahhh, I got it in the pom for kundera-cassandra-2.0.7.1 it has a dependency
on 2.0.8.SNAPSHOT

<dependency>
<groupId>com.impetus.core</groupId>
<artifactId>kundera-core</artifactId>
<version>2.0.8-SNAPSHOT</version>
</dependency>

Mark

**Comment by bytor99999** [comment]
Also, I noticed in core. the pom has a dependency on Hibernate and
Hibernate version 3.6.8. And in our project for the rdbms persistence unit
using pure JPA we are using Hibernate 4.1.0.FINAL. Will this cause any
conflicts or problems too?

Thanks

Mark

**Comment by bytor99999** [comment]
Also looks like I know have to also have a dependency in my project to
core, which I didn't need before. I got away with just kundera-cassandra
before.

Mark

**Comment by bytor99999** [comment]
So I am getting past that error now. So it looks like the fix works.
However, I do get another error

Caused by: java.lang.IllegalArgumentException: Entity object is invalid,
operation failed. Please check previous log message for details
at
com.impetus.kundera.persistence.PersistenceDelegator.persist(PersistenceDelegator.java:137)

So that looks like everything is running, but now when I try to persist an
object it fails here. Not sure why. I think I have the @Table configured
correctly and I am sure I have the columnFamily and keyspace in Cassandra.
But the error doesn't give much more details, even in the previous log
messages.

2012-08-23 13:57:53,704 [main] INFO  org.scale7.cassandra.pelops.Pelops -
Pelops adds new pool 10.0.30.74:9160:testing
2012-08-23 13:57:53,707 [main] WARN
 com.impetus.kundera.classreading.ClasspathReader - Error during loading
from context , Caused by:error in opening zip file
2012-08-23 13:57:53,707 [main] WARN
 com.impetus.kundera.classreading.ClasspathReader - Error during loading
from context , Caused by:error in opening zip file
2012-08-23 13:57:53,714 [main] WARN
 com.impetus.kundera.classreading.ClasspathReader - Error during loading
from context , Caused by:error in opening zip file
2012-08-23 13:57:53,714 [main] WARN
 com.impetus.kundera.classreading.ClasspathReader - Error during loading
from context , Caused by:error in opening zip file
2012-08-23 13:57:53,866 [main] INFO
 com.impetus.kundera.persistence.EntityManagerFactoryImpl -
EntityManagerFactory created for persistence unit : cassandra_pu
2012-08-23 13:57:53,924 [main] INFO
 com.impetus.kundera.client.ClientResolver - Returning client instance
for:cassandra_pu

Thanks

Mark

**Comment by bytor99999** [comment]
OK, so I think this is the actual error

2012-08-23 14:37:49,370 [main] ERROR
com.impetus.kundera.persistence.PersistenceValidator - Entity to be
persisted can't have Primary key set to null.

This is a new instance of the domain object so therefore it is wouldn't
have an id created yet. Should ID be set to @GeneratedValue?

Thanks

Mark

**Comment by bytor99999** [comment]
Nevermind on the generation of ids. Still old school single RDBMS database
thinking. I have to generate it as a UUID.

Mark

**Comment by bytor99999** [comment]
So, after all that, I changed the id to use a UUID, and the id field is now a String. But now back to the original error. I get the other version of the error I was getting when I opened this issue and that was a NullPointerException because the MetaData is now null again.

"...then the two classes I list in the xml aren't picked up and no MetaData is created for them, even though I correctly put them in the persistence.xml"

Thanks

Mark
